src/test/java/org/owasp/html/HtmlLexerTest.java

```diff
@@ -121,6 +121,31 @@ public static final void testShortTags() {
         "TAGEND: >");
   }
 
+  @Test
+  public static final void testCommentDeclarationWith0CommentsAndXss() throws Exception
+  {
+    //check https://datatracker.ietf.org/doc/html/rfc1866#section-3.2.5
+    assertTokens("<!><img src=1 onError=alert(\"nice\")>",
+            "COMMENT: <!>",
+            "TAGBEGIN: <img",
+            "ATTRNAME: src",
+            "ATTRVALUE: 1",
+            "ATTRNAME: onError",
+            "ATTRVALUE: alert(\"nice\")",
+            "TAGEND: >"
+    );
+  }
+
+  @Test
+  public static final void testCommentDeclarationWith0CommentsAndTag() throws Exception
+  {
+    assertTokens("<!--><img>",
+            "COMMENT: <!-->",
+            "TAGBEGIN: <img",
+            "TAGEND: >"
+    );
+  }
```

Comment:
Thanks for the tests.

Could you add tests for 1 and 3 dashes?
They needn't be 3 tests.
You could probably incorporate them into this test by appending

    a<!--->b<!->c